### DISCUSSION TITLE
Clarification needed: 'Simplify' Transfer Learning vs. 'DP-Gen Run' Finetuning for potential model updates

### DISCUSSION THREAD

**Discussion by aersilang** [deepmodeling/dpgen#1964] [Q&A]
### Summary

Hello,

I have recently encountered a challenge regarding potential function development. I currently possess a potential model that accurately simulates the Phase A and Phase B of a specific alloy. I now intend to incorporate data for Phase C into this potential. Please note that no new elements are being introduced, but my current dataset for Phase C is quite limited. Consequently, I am looking to utilize transfer learning.

During my research, I identified two possible workflows, but I am confused about their practical differences:

The "Simplify" Method ((https://docs.deepmodeling.com/projects/dpgen/en/latest/simplify/simplify.html)): Upon testing, it appears this method is primarily designed for database screening or selection rather than direct model updating.

The "DP-Gen Run" Method (https://github.com/deepmodeling/deepmd-kit/discussions/4580): This seems to behave more like a standard model refitting process.

I have consulted technical support, but they were unable to provide a definitive explanation of the distinctions between these two paths. I would appreciate your guidance on the following:

Which method is the most efficient and "convenient" path for adding a small amount of new phase data to an existing model?

Which JSON configuration template should be used for the most up-to-date workflow?

I would be extremely grateful for any insights or documentation you could provide to help resolve this.

### DeePMD-kit Version

DPGEN

### Backend and its version

all time

### Details

that‘s all

**Comment by dosubot** [comment]
Hi @aersilang! I'm [Dosu](https://go.dosu.dev/dosubot) and I’m helping the deepmd-kit team.


Your understanding is correct! Let me clarify the key differences:

**Simplify ≠ Transfer Learning**: The [DPGEN Simplify workflow](https://docs.deepmodeling.com/projects/dpgen/en/latest/simplify/simplify.html) is designed for **data screening and selection**, not model updating. It filters representative frames from large datasets based on model deviation—useful for dataset curation, but not for adding new phase data to an existing model.

**For your use case (adding Phase C to existing Phase A+B model)**, use the **DP-Gen Run workflow** with transfer learning parameters:

```json
{
  "training_finetune_model": [
    "phaseAB_model_0.pb",
    "phaseAB_model_1.pb",
    "phaseAB_model_2.pb",
    "phaseAB_model_3.pb"
  ],
  "training_reuse_old_ratio": "auto:5",
  "training_reuse_start_lr": 0.0001,
  "training_reuse_start_pref_e": 0.1,
  "training_reuse_start_pref_f": 100,
  "init_data_sys": [
    "path/to/phaseA_data",
    "path/to/phaseB_data", 
    "path/to/phaseC_data"
  ]
}
```

Key parameters:
- [`training_finetune_model`](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html): Points to your existing Phase A+B frozen models (.pb files)
- [`training_reuse_old_ratio`](https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html): Balances old vs new data to prevent catastrophic forgetting (default practice uses ~0.9, meaning 90% weight on original data)

**Regarding `--finetune` vs `--init-model`**: These are [functionally equivalent for model retraining](https://github.com/deepmodeling/deepmd-kit/discussions/4580)—the only difference is file format (frozen `.pb` vs checkpoint `.ckpt`). For DPGEN workflows, `training_finetune_model` uses the `--finetune` flag internally.

**Best practices for small Phase C datasets**:
1. Start with a lower learning rate (e.g., 1e-4)
2. Include Phase A+B data in `init_data_sys` alongside Phase C
3. Use `training_reuse_old_ratio` to maintain performance on original phases



*To reply, just mention [@dosu](https://go.dosu.dev/dosubot).*

---
<sup>How did I do? [Good](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=great_response) | [Irrelevant](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=irrelevant_answer) | [Incorrect](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=incorrect_sources) | [Verbose](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=too_verbose) | [Hallucination](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=hallucination) | [Report 🐛](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=bug_report) | [Other](https://app.dosu.dev/response-feedback/4c75ed6e-f1e8-4ac4-b222-9edd3254bbd4?feedback_type=other)</sup>

[![Open in Cursor](https://dosu.dev/dosu-in-cursor.svg)](https://cursor.com/link/prompt?text=Your%20understanding%20is%20correct%21%20Let%20me%20clarify%20the%20key%20differences%3A%0A%0A%2A%2ASimplify%20%E2%89%A0%20Transfer%20Learning%2A%2A%3A%20The%20%5BDPGEN%20Simplify%20workflow%5D%28https%3A//docs.deepmodeling.com/projects/dpgen/en/latest/simplify/simplify.html%29%20is%20designed%20for%20%2A%2Adata%20screening%20and%20selection%2A%2A%2C%20not%20model%20updating.%20It%20filters%20representative%20frames%20from%20large%20datasets%20based%20on%20model%20deviation%E2%80%94useful%20for%20dataset%20curation%2C%20but%20not%20for%20adding%20new%20phase%20data%20to%20an%20existing%20model.%0A%0A%2A%2AFor%20your%20use%20case%20%28adding%20Phase%20C%20to%20existing%20Phase%20A%2BB%20model%29%2A%2A%2C%20use%20the%20%2A%2ADP-Gen%20Run%20workflow%2A%2A%20with%20transfer%20learning%20parameters%3A%0A%0A%60%60%60json%0A%7B%0A%20%20%22training_finetune_model%22%3A%20%5B%0A%20%20%20%20%22phaseAB_model_0.pb%22%2C%0A%20%20%20%20%22phaseAB_model_1.pb%22%2C%0A%20%20%20%20%22phaseAB_model_2.pb%22%2C%0A%20%20%20%20%22phaseAB_model_3.pb%22%0A%20%20%5D%2C%0A%20%20%22training_reuse_old_ratio%22%3A%20%22auto%3A5%22%2C%0A%20%20%22training_reuse_start_lr%22%3A%200.0001%2C%0A%20%20%22training_reuse_start_pref_e%22%3A%200.1%2C%0A%20%20%22training_reuse_start_pref_f%22%3A%20100%2C%0A%20%20%22init_data_sys%22%3A%20%5B%0A%20%20%20%20%22path/to/phaseA_data%22%2C%0A%20%20%20%20%22path/to/phaseB_data%22%2C%20%0A%20%20%20%20%22path/to/phaseC_data%22%0A%20%20%5D%0A%7D%0A%60%60%60%0A%0AKey%20parameters%3A%0A-%20%5B%60training_finetune_model%60%5D%28https%3A//docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html%29%3A%20Points%20to%20your%20existing%20Phase%20A%2BB%20frozen%20models%20%28.pb%20files%29%0A-%20%5B%60training_reuse_old_ratio%60%5D%28https%3A//docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html%29%3A%20Balances%20old%20vs%20new%20data%20to%20prevent%20catastrophic%20forgetting%20%28default%20practice%20uses%20~0.9%2C%20meaning%2090%25%20weight%20on%20original%20data%29%0A%0A%2A%2ARegarding%20%60--finetune%60%20vs%20%60--init-model%60%2A%2A%3A%20These%20are%20%5Bfunctionally%20equivalent%20for%20model%20retraining%5D%28https%3A//github.com/deepmodeling/deepmd-kit/discussions/4580%29%E2%80%94the%20only%20difference%20is%20file%20format%20%28frozen%20%60.pb%60%20vs%20checkpoint%20%60.ckpt%60%29.%20For%20DPGEN%20workflows%2C%20%60training_finetune_model%60%20uses%20the%20%60--finetune%60%20flag%20internally.%0A%0A%2A%2ABest%20practices%20for%20small%20Phase%20C%20datasets%2A%2A%3A%0A1.%20Start%20with%20a%20lower%20learning%20rate%20%28e.g.%2C%201e-4%29%0A2.%20Include%20Phase%20A%2BB%20data%20in%20%60init_data_sys%60%20alongside%20Phase%20C%0A3.%20Use%20%60training_reuse_old_ratio%60%20to%20maintain%20performance%20on%20original%20phases)&nbsp;[![Join Discord](https://img.shields.io/badge/join-5865F2?logo=discord&logoColor=white&label=)](https://go.dosu.dev/discord-bot)&nbsp;[![Share on X](https://img.shields.io/badge/X-share-black)](https://twitter.com/intent/tweet?text=%40dosu_ai%20helped%20me%20solve%20this%20issue!&url=https%3A//github.com/deepmodeling/deepmd-kit/issues/5279)

**Comment by aersilang** [comment]
However, the documentation(https://tutorials.deepmodeling.com/en/latest/CaseStudies/Transfer-learning/Transfer-learning.html) at  describes it as follows: "This tutorial will introduce how to implement potential energy surface (PES) transfer-learning by using the [DP-GEN](https://github.com/deepmodeling/dpgen) software. In [DP-GEN](https://github.com/deepmodeling/dpgen) (version > 0.8.0), the “simplify” module is designed for this purpose. Suppose that we have completed a typical [DP-GEN](https://github.com/deepmodeling/dpgen) flow, and obtained the DFT dataset and four DP models. The workflow of “simplify” is similar to a typical [DP-GEN](https://github.com/deepmodeling/dpgen) process: iteratively training the DP models with the (re-) labeled data (00.train), picking data according to prediction deviations between different models (01.model_devi), and (re-) labeling the picked data (02.fp). Repeat the iterations until convergence is achieved. Then, the relabeled new dataset that is sufficient to refine the DP model is successfully collected." So, which one is actually the correct approach?

**Comment by wanghan-iapcm** [comment]
"Simplify" : active learning to reduce the size of an **existing** training data set.
"DP-Gen Run": concurrent learning to generate a training dataset **from scratch**. the issue you mentioned (https://github.com/deepmodeling/deepmd-kit/discussions/4580) seems to be not very relevant. 

Reference for dpgen concurrent learning: https://doi.org/10.1016/j.cpc.2020.107206

**Comment by aersilang** [comment]
Regarding the DP-Gen run example, I noticed a supplement on the (https://docs.deepmodeling.com/projects/dpgen/en/latest/run/param.html) with the following description:

'type: list[str], optional
argument path: run_jdata[mlp_engine=dp]/training_finetune_model
At interation 0, finetune the model parameters from the given frozen models. Number of element should be equal to numb_models'

Based on this, I have two specific questions:

Can the simplify method still be used for transfer learning? If so, could you please point me to the latest JSON configuration files and documentation? Or is it now strictly for database simplification (implying the functionality described in https://tutorials.deepmodeling.com/en/latest/CaseStudies/Transfer-learning/Transfer-learning.html  has been deprecated)?

If DP-Gen run is the only way to achieve my goal, is there a recommended procedure to perform finetuning using an existing potential function (frozen model) or a checkpoint file?

I would be extremely grateful for your guidance on these matters.

[a comment by an automated account is not shown]
